### DISCUSSION TITLE
function() in the scheduler

### DISCUSSION THREAD

**Discussion by SoftVienka** [emsesp/EMS-ESP32#2111] [Q&A] [answered]
Hello
Is it possible to use the round() function in the scheduler?

Write entity RAM float (17.25) to entity RAM as int (17)

example:
Command: custom/interval
Value: (x > 2 && x < 10) ? custom/interval_1 : custom/interval

interval_1 is a float and I want it to be stored as an int into interval

when I write round(custom/interval1) then npr: round17.25 is written into the interval

**Comment by proddy** [comment]
@MichaelDvP we could set a value type to the RAM custom entity. Now it's fixed as DeviceValueType::STRING in WebCustomEntity::update(). And do the conversion in the C++ code to BOOL *INT* STRING

Adding functions to the command/scheduler shunting Yard will be a big change. This is what is supported today: https://emsesp.org/Configuring/#scheduling-actions

**Reply by MichaelDvP** [reply]
I think adding a few unary functions to shunting yard is not so much work, I think `round`, `int`, `abs` are usefull. Maybe also `sqrt`.  I think trigonometric, logarithmic and hyperbolic function are rarly used in heating, will skip them. I'll make a change to my dev next days and a PR after testing.
The ram values are used for commands which always expects `const char*`arguments, also shunting yard works with strings. But we never add uom to strings, maybe it's nice to show a uom to ram values?

**Reply by MichaelDvP** [reply, marked as answer]
PS: `round` should work now in https://github.com/MichaelDvP/EMS-ESP32/releases/tag/latest

**Reply by proddy** [reply]
nice! I created a new issue for this, for 3.7.0

**Reply by SoftVienka** [reply]
round() works, it immediately looks prettier

just a side note:
In the dashboard, every other row could be highlighted, as it is in devices.

Thank you
